src/zkevm_specs/evm_circuit/execution/error_oog_memory_copy.py

```diff
@@ -0,0 +1,69 @@
+from zkevm_specs.evm_circuit.table import CallContextFieldTag, RW
+from zkevm_specs.util.param import (
+    GAS_COST_ACCOUNT_COLD_ACCESS,
+    GAS_COST_FASTEST,
+    GAS_COST_WARM_ACCESS,
+    N_BYTES_MEMORY_ADDRESS,
+)
+from ...util import FQ
+from ..instruction import Instruction
+from ..opcode import Opcode
+from ...util import N_BYTES_GAS
+
+
+def error_oog_memory_copy(instruction: Instruction):
+    # retrieve op code associated to oog constant error
+    opcode = instruction.opcode_lookup(True)
+    (
+        is_calldata_copy,
+        is_code_copy,
+        is_ext_code_copy,
+        is_return_data_copy,
+    ) = instruction.multiple_select(
+        opcode, (Opcode.CALLDATACOPY, Opcode.CODECOPY, Opcode.EXTCODECOPY, Opcode.RETURNDATACOPY)
+    )
+
+    # Constrain opcode must be one of CALLDATACOPY, CODECOPY, EXTCODECOPY or RETURNDATACOPY.
+    instruction.constrain_equal(
+        is_calldata_copy + is_code_copy + is_ext_code_copy + is_return_data_copy, FQ(1)
+    )
+
+    # EXTCODECOPY has one more extra stack pop for external address
+    stack_offset = 0
+    if is_ext_code_copy == FQ(1):
+        external_address = instruction.stack_lookup(RW.Read, stack_offset)
+        stack_offset += 1
+    memory_offset_word = instruction.stack_lookup(RW.Read, stack_offset)
+    copy_size_word = instruction.stack_lookup(RW.Read, stack_offset + 2)
+
+    # Get constant gas cost
+    if is_ext_code_copy == FQ(1):
+        address = instruction.word_to_fq(external_address, N_BYTES_MEMORY_ADDRESS)
+        tx_id = instruction.call_context_lookup(CallContextFieldTag.TxId)
+        is_warm = instruction.add_account_to_access_list(
+            tx_id, address, instruction.reversion_info()
+        )
```

Comment:
Since we know it's going to be reverted, we can just read here to save some read/write.
```suggestion
        is_warm = instruction.read_account_to_access_list(tx_id, address)
```